File: Search/ECMSearchWPF/app_code/clsHive.vb
' ***********************************************************************
' Assembly         : ECMSearchWPF
' Author           : wdale
' Created          : 07-16-2020
'
' Last Modified By : wdale
' Last Modified On : 07-16-2020
' ***********************************************************************
' <copyright file="clsHive.vb" company="D. Miller and Associates, Limited">
'     Copyright @ DMA Ltd 2020 all rights reserved.
' </copyright>
' <summary></summary>
' ***********************************************************************
Imports ECMEncryption

''' <summary>
''' Class clsHive.
''' </summary>
Public Class clsHive

    ''' <summary>
    ''' The log
    ''' </summary>
    Dim LOG As New clsLogMain
    'Dim proxy As New SVCSearch.Service1Client
    'Dim EP As New clsEndPoint
    ''' <summary>
    ''' The en c2
    ''' </summary>
    Dim ENC2 As New ECMEncrypt()



    ''' <summary>
    ''' Updates the name of the hive server.
    ''' </summary>
    ''' <param name="TableName">Name of the table.</param>
    Sub updateHiveServerName(ByVal TableName As String)
        Dim S As String = "Update " + TableName + " set RepoSvrName = (select @@SERVERNAME) where RepoName is null "
        S = ENC2.AES256EncryptString(S)
        Dim BB As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, S, _UserID, ContractID)
        gLogSQL(BB, ENC2.AES256EncryptString(S))

    End Sub
    ''' <summary>
    ''' Updates the name of the hive repo.
    ''' </summary>
    ''' <param name="TableName">Name of the table.</param>
    Sub updateHiveRepoName(ByVal TableName As String)
        Dim S As String = "Update " + TableName + " set RepoSvrName = (SELECT DB_NAME() AS DataBaseName) where RepoName is null "
        'Dim proxy As New SVCSearch.Service1Client

        'EP.setSearchSvcEndPoint(proxy)
        S = ENC2.AES256EncryptString(S)
        Dim BB As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, S, _UserID, ContractID)
        gLogSQL(BB, ENC2.AES256EncryptString(S))

    End Sub

    ''' <summary>
    ''' Allows an object to try to free resources and perform other cleanup operations before it is reclaimed by garbage collection.
    ''' </summary>
    Protected Overrides Sub Finalize()
        Try
        Finally
            MyBase.Finalize()      'define the destructor
            GC.Collect()
            GC.WaitForPendingFinalizers()
        End Try
    End Sub

End Class

File: Search/ECMSearchWPF/CS/app_code/clsHive.cs
// VBConversions Note: VB project level imports
using System.Windows.Input;
using System.Windows.Data;
using System.Windows.Documents;
using System.Xml.Linq;
using System.Collections.Generic;
using System.Windows.Navigation;
using System.Windows.Media.Imaging;
using System.Windows;
using Microsoft.VisualBasic;
using System.Windows.Media;
using System.Collections;
using System;
using System.Windows.Shapes;
using System.Windows.Controls;
using System.Threading.Tasks;
using System.Linq;
using System.Diagnostics;
// End of VB project level imports


namespace ECMSearchWPF
{
	public class clsHive
	{
		
		clsLogging LOG = new clsLogging();
		//Dim proxy As New SVCSearch.Service1Client
		//Dim EP As New clsEndPoint
		clsEncryptV2 ENC2 = new clsEncryptV2();
		
		
		
		public void updateHiveServerName(string TableName)
		{
			string S = "Update " + TableName + " set RepoSvrName = (select @@SERVERNAME) where RepoName is null ";
			S = ENC2.EncryptPhrase(S, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, S, GLOBALS._UserID, GLOBALS.ContractID);
		}
		public void updateHiveRepoName(string TableName)
		{
			string S = "Update " + TableName + " set RepoSvrName = (SELECT DB_NAME() AS DataBaseName) where RepoName is null ";
			//Dim proxy As New SVCSearch.Service1Client
			
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted += new System.EventHandler(client_ExecuteSqlNewConn);
			//EP.setSearchSvcEndPoint(proxy)
			S = ENC2.EncryptPhrase(S, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, S, GLOBALS._UserID, GLOBALS.ContractID);
		}
		public void client_ExecuteSqlNewConn(object sender, SVCSearch.ExecuteSqlNewConnSecureCompletedEventArgs e)
		{
			if (e.Error == null)
			{
				bool B = System.Convert.ToBoolean(e.Result);
			}
			else
			{
				string ErrSql = (string) e.MySql;
				modGlobals.gErrorCount++;
				LOG.WriteToSqlLog((string) ("ERROR clsLIBRARYUSERS 100: " + e.Error.Message));
				LOG.WriteToSqlLog((string) ("ERROR clsLIBRARYUSERS 100: " + ErrSql));
			}
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted -= new System.EventHandler(client_ExecuteSqlNewConn);
		}
		~clsHive()
		{
			try
			{
			}
			finally
			{
				base.Finalize(); //define the destructor
				GC.Collect();
				GC.WaitForPendingFinalizers();
			}
		}
		
	}
	
}
